source/remoteClient/input.py

```diff
@@ -0,0 +1,152 @@
+# A part of NonVisual Desktop Access (NVDA)
+# Copyright (C) 2015-2025 NV Access Limited, Christopher Toth, Tyler Spivey, Babbage B.V., David Sexton and others.
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+
+import ctypes
+from ctypes import POINTER, Structure, Union, c_long, c_ulong, wintypes
+
+import api
+import baseObject
+import braille
+import brailleInput
+import globalPluginHandler
+import scriptHandler
+import vision
+
+INPUT_MOUSE = 0
+INPUT_KEYBOARD = 1
+INPUT_HARDWARE = 2
+MAPVK_VK_TO_VSC = 0
+KEYEVENTF_EXTENDEDKEY = 0x0001
+KEYEVENTF_KEYUP = 0x0002
+KEYEVENT_SCANCODE = 0x0008
+KEYEVENTF_UNICODE = 0x0004
+
+
+class MOUSEINPUT(Structure):
+	_fields_ = (
+		("dx", c_long),
+		("dy", c_long),
+		("mouseData", wintypes.DWORD),
+		("dwFlags", wintypes.DWORD),
+		("time", wintypes.DWORD),
+		("dwExtraInfo", POINTER(c_ulong)),
+	)
+
+
+class KEYBDINPUT(Structure):
+	_fields_ = (
+		("wVk", wintypes.WORD),
+		("wScan", wintypes.WORD),
+		("dwFlags", wintypes.DWORD),
+		("time", wintypes.DWORD),
+		("dwExtraInfo", POINTER(c_ulong)),
+	)
+
+
+class HARDWAREINPUT(Structure):
+	_fields_ = (
+		("uMsg", wintypes.DWORD),
+		("wParamL", wintypes.WORD),
+		("wParamH", wintypes.WORD),
+	)
+
+
+class INPUTUnion(Union):
+	_fields_ = (
+		("mi", MOUSEINPUT),
+		("ki", KEYBDINPUT),
+		("hi", HARDWAREINPUT),
+	)
+
+
+class INPUT(Structure):
+	_fields_ = (
+		("type", wintypes.DWORD),
+		("union", INPUTUnion),
+	)
+
+
+class BrailleInputGesture(braille.BrailleDisplayGesture, brailleInput.BrailleInputGesture):
+	def __init__(self, **kwargs):
+		super().__init__()
+		for key, value in kwargs.items():
+			setattr(self, key, value)
+		self.source = "remote{}{}".format(self.source[0].upper(), self.source[1:])
```

Comment:
Where is `source` coming from? What if it's not defined?
Also can we use an f-string here?
```suggestion
		self.source = "remote{self.source[0].upper()}{self.source[1:]}"
```

Reply:
That's an excellent question :) @LeonarddeR  Do you have any idea?